src/rules/no-redundant-publishConfig.ts

```diff
@@ -0,0 +1,81 @@
+import type { AST as JsonAST } from "jsonc-eslint-parser";
+
+import { fixRemoveObjectProperty } from "eslint-fix-utils";
+import * as ESTree from "estree";
+
+import { createRule } from "../createRule.ts";
+import { isJSONStringLiteral } from "../utils/predicates.ts";
+
+export const rule = createRule({
+	create(context) {
+		let packageName: string | undefined;
+		let publishConfigAccessProperty: JsonAST.JSONProperty | undefined;
+
+		return {
+			"Program > JSONExpressionStatement > JSONObjectExpression > JSONProperty[key.value=name]"(
+				node: JsonAST.JSONProperty,
+			) {
+				if (isJSONStringLiteral(node.value)) {
+					packageName = node.value.value;
+				}
+			},
+			"Program > JSONExpressionStatement > JSONObjectExpression > JSONProperty[key.value=publishConfig]"(
+				node: JsonAST.JSONProperty,
+			) {
+				if (node.value.type !== "JSONObjectExpression") {
+					return;
+				}
+
+				for (const property of node.value.properties) {
+					if (
+						isJSONStringLiteral(property.key) &&
+						property.key.value === "access"
+					) {
+						publishConfigAccessProperty = property;
+						break;
+					}
+				}
+			},
+			"Program:exit"() {
+				if (!packageName || !publishConfigAccessProperty) {
+					return;
+				}
+
+				const isScopedPackage = packageName.startsWith("@");
+
+				if (!isScopedPackage) {
+					context.report({
+						messageId: "redundantAccess",
+						node: publishConfigAccessProperty,
+						suggest: [
+							{
+								fix: fixRemoveObjectProperty(
+									context,
+									publishConfigAccessProperty as unknown as ESTree.Property,
+								),
+								messageId: "removeAccess",
+							},
+						],
+					});
+				}
+			},
+		};
+	},
+	meta: {
+		docs: {
+			category: "Best Practices",
+			description:
+				"Warns when publishConfig.access is used in unscoped packages.",
+			recommended: true,
+		},
+		hasSuggestions: true,
+		messages: {
+			redundantAccess:
+				"'publishConfig.access' is redundant for unscoped packages - they are always public.",
+			removeAccess: "Remove the redundant 'access' field.",
+		},
+		schema: [],
+		type: "suggestion",
+	},
+	name: "no-redundant-publishConfig",
```

Comment:
I'm a little confused by the camelCase in the rule name. What do you think about renaming it to `no-redundant-publish-config`?

Reply:
The property name is `publishConfig`.  so, for all of our rules that are geared around specific properties, we've adopted the convention of using the property name as a single "word" in the rule name.  e.g. `require-publishConfig` = require the `publishConfig` property.  If the rule name were `require-publish-config` that would not be as clear.